react/week-3/learning-objectives.md

```diff
@@ -1,22 +1,23 @@
 ## Learning Objectives - React Week 3
 
-### Lifecycle
+### Data fetching (advanced)
 
-- "Clean up" a component when it is unmounted
-
-### Data fetching
-
-- Show some data from an API in a component
-  - Understand that `render` will be called before data is available and handle this appropriately
-  - Use `componentDidMount` lifecycle method to trigger `fetch()`
-  - Be able to hold data, loading & error states and render correctly based on these
+- Can explain what the effect of unmounting a component
+- Cleaning up effects
+  - Recognise when an effect could cause a memory leak
+  - Be able to prevent an effect causing a memory leak using the unsubscribe function of `useEffect()`
+- Allow effects to update in response to prop changes
+  - Explain why a component with an effect dependent on props is broken with empty `useEffect()` dependencies (`[]`)
+  - Be able to fix a component with an effect dependent on props using the `useEffect()` dependencies
```

Comment:
We're using the words "broken" and "fix" in the two points above - I haven't seen `useEffect` described that way before. Could we reword slightly? Suggestion below but by no means perfect.
- "explain how we can define that something (like fetching data for example) should happen only once when a component loads, by setting `useEffect`'s dependency to an empty array: `, []`".
- "explain how we can define that something (like a calculation for example) should happen each time a specific prop is updated, by setting `useEffect`'s dependency to that prop's name in square brackets: `, [specificProp]`".

Reply:
We could potentially add something around the fact that there can be multiple useEffects in a component.
e.g. "can explain that `useEffect` can be used several times inside a component to respond to different triggers"

Reply:
Sorry @40thieves I've just realised we already include the `[]` option in week 2's data fetching, makes more sense now! 🤦‍♀️

Reply:
The point I was attempting to make was that students know how to use `useEffect` (the double "use" really bothers me 😂) with no deps as you say. Then I was going to show them a situation where this doesn't work (i.e. is "broken"), discuss why it doesn't work and finally show them how to "fix" it.

But I agree that "broken" and "fix" aren't the best words to use here. I'll go with something like your wording, thanks 👍